test/snippet/core/algorithm/configuration.cpp

```diff
@@ -1,88 +1,96 @@
+#include <iostream>
+
+//! [configuration_setup]
 #include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
 
 using namespace seqan3;
 
-struct bar
+enum struct my_id : int
 {
-    float value = 0;
+    bar_id,
+    foo_id
 };
 
-struct foo
+struct bar : public pipeable_config_element
 {
-    int value = 0;
+    static constexpr my_id id{my_id::bar_id};
+
+    bar()                        = default;
```

Comment:
can't you implement the default constructors in `pipeable_config_element` and then just `using` them here?

Reply:
pipeable is an empty base class. In general I would have loved to keep the config elements aggregate types but, then they become ugly to initialize because you need to give an empty construciton argument for the empty base class.

Reply:
Ok, while thinking about it I was making my live more complicated than it had to. I changed the base class to a crtp base. Further the class is now an aggregate so that in general configs are agregates to. This solved all boilerplate code with the constructors and such.

Reply:
nice